sr_unix/wait_for_disk_space.c

```diff
@@ -161,7 +164,8 @@ void wait_for_disk_space(sgmnt_addrs *csa, char *fn, int fd, off_t offset, char
 	/* Only report if we were the process to set the current freeze comment; otherwise someone else reported it. */
 	if (STRCMP(wait_comment, jnlpool->jnlpool_ctl->freeze_comment) == 0)
 	{
-		CLEAR_ANTICIPATORY_FREEZE(freeze_cleared);			/* sets freeze_cleared */
+		freeze_cleared = FALSE;						/* Default */
+		CLEAR_ANTICIPATORY_FREEZE(freeze_cleared);			/* May reset freeze_cleared */
```

Comment:
It seems a safer change to fix the CLEAR_ANTICIPATORY_FREEZE macro to set "freeze_cleared" to FALSE in case it does not do the clear. This should address other callers of this macro too (e.g. jnl_write_attempt.c). Surprised that did not show up in the scan alert whereas this module showed up.

Reply:
Since you are telling me that is the proper change, I'll make it but I have had nothing to do with the freeze logic and know next to nothing about it.

If the same issue exists in the other routines, I'm only mildly surprised they weren't flagged. It seems the scanner runs "scenarios" making decisions are each "if" only validating the code it reaches. It probably tries to do enough scenarios that it gets MOST of the code to run but I'm unsurprised there are gaps. There has to be a limit on generated scenarios else it will run for years.

Reply:
The CLEAR_ANTICIPATORY_FREEZE macro has been changed to set freeze_cleared on return in all cases. But line 167 above still needs to be removed.